ROADMAP.md

```diff
@@ -0,0 +1,123 @@
+  The current documentation is not up-to-date which makes using xk6-browser outside of the examples quite difficult.
+
+  *How will we achieve this?*<br>
+  We are prioritizing working on documentation in all upcoming cycles.
+
+  *Definition of Done*<br>
+  When the documentation reflects the state of the current API.
+
+
+Mid-term goals
+--------------
+
+These are goals achievable within 6-12 months, tentatively done by mid 2023.
+
+- **Transition our API to be async/`Promise` based.**<br>
+  Currently (April 2022), most of our API is synchronous. This is due to the historical fact that k6 didn't support async behavior because of a missing per VU event loop, which is [now available](https://github.com/grafana/k6/pull/2228).
+  Async APIs are important for a browser testing tool, since most browser behavior and [CDP](https://chromedevtools.github.io/devtools-protocol/) (the protocol we use to communicate with the browser) is event-based. We need to expose an async API to implement this missing functionality and reach feature parity with tools like Playwright or Puppeteer.
+
+  *How will we achieve this?*<br>
+  By gradually transitioning the current API to an async implementation, and implementing new features as async.
+
+  *Definition of Done*<br>
+  When most of the API can be used asynchronously.
+
+
+- **Beta availability of the extension in k6 Cloud for all users.**<br>
+
+  *How will we achieve this?*<br>
+  The deployment should be optimized and the extension thoroughly tested before making it available to all users. Frontend changes should be done at this point, and usage costs (CPU, RAM, storage) and pricing details should be determined, followed by public announcements of the availability. Features such as screen capture, video recording, downloading, and uploading files should be available.
+
+  *Definition of Done*<br>
+  When all users are eligible to upgrade to a plan that includes browser testing.
+
+
+- **Increase test code coverage; refactor problematic areas of the codebase; fix "flaky" tests, linter issues, etc.**<br>
+  This is both a mid and long term goal we'll be constantly focusing on. See issues [#228](https://github.com/grafana/xk6-browser/issues/228), [#241](https://github.com/grafana/xk6-browser/issues/241) and [#58](https://github.com/grafana/xk6-browser/issues/58).
+
+  *How will we achieve this?*<br>
+  By gradually adding more unit and functional tests and fixing issues when the opportunity arises. We won't have long stints where we focus exclusively on this goal, but it's something we'll dedicate as much time as possible to.
+
+  *Definition of Done*<br>
+  The goal is not to reach 100% code coverage, but, realistically speaking, 80 or 90% should be achievable. Our CI test runs should be stable and flake-free, and we should have no linter issues.
+
+
+Long-term goals
+---------------
+
+These are goals achievable after a year, tentatively done in 2024 and beyond.
```

Comment:
> tentatively done in 2024 and beyond.

I would not include timelines that are 2 years away. Let's leave the "long-term goals" without dates. Things will change so we can't make predictions that far in the future.

Reply:
Indeed. Changed in d17ee93.